NASA-IMPACT/nasa-apt-frontend
# Replace input/output variables list with a table

See https://github.com/NASA-IMPACT/nasa-apt/issues/849.

List of changes:

- Created a common table component for variables in a separate file.
- Replaced variable list with the table component in `document-body` and `journal-pdf-preview` files.
- Refactored the types definition file for readability.

How to test:

- Visit a document with no I/O variables; it should render "No content available" in the respective sections.
- Add I/O variables to a document and go to its view page to confirm the variables are displayed as tables.
- Visit the PDF preview route for that document ([example](http://localhost:9000/documents/testing-the-journal-pdf-output-w/v1.0/journal-pdf-preview)) and confirm tables are rendered.

Please note the "Download" button will generate the sections in the previous format and will only change after a release.

@wrynearson this is ready for review.
